libs/pika/topology/src/topology.cpp

```diff
@@ -1579,34 +1576,32 @@ namespace pika { namespace threads {
     }
 
     ///////////////////////////////////////////////////////////////////////////
-    topology& create_topology()
+    topology& create_topology() noexcept
     {
         static topology topo;
         return topo;
     }
 
     ///////////////////////////////////////////////////////////////////////////
-    namespace detail {
-        struct hw_concurrency
-        {
-            hw_concurrency()
+    struct hw_concurrency
+    {
+        hw_concurrency()
```

Comment:
This should maybe be noexcept... unsure about the android call, but I would guess `hwloc_hardware_concurrency` is a C function.